Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -2825,40 +2825,31 @@ WhiteSpace RenderStyle::whiteSpace() const
     return WhiteSpace::Normal;
 }
 
-void RenderStyle::setWordSpacing(Length&& value)
+void RenderStyle::setLetterSpacing(Length&& spacing)
 {
-    float fontWordSpacing;
-    switch (value.type()) {
-    case LengthType::Auto:
-        fontWordSpacing = 0;
-        break;
-    case LengthType::Percent:
-        fontWordSpacing = value.percent() * fontCascade().widthOfSpaceString() / 100;
-        break;
-    case LengthType::Fixed:
-        fontWordSpacing = value.value();
-        break;
-    case LengthType::Calculated:
-        fontWordSpacing = value.nonNanCalculatedValue(static_cast<float>(maxValueForCssLength));
-        break;
-    default:
-        ASSERT_NOT_REACHED();
-        fontWordSpacing = 0;
-        break;
+    if (fontCascade().computedLetterSpacing() == spacing)
+        return;
+
+    bool oldShouldDisableLigatures = fontDescription().shouldDisableLigaturesForSpacing();
+    m_inheritedData.access().fontCascade.setLetterSpacing(WTFMove(spacing));
+
+    // Switching letter-spacing between zero and non-zero requires updating fonts (to enable/disable ligatures)
+    if (oldShouldDisableLigatures != bool(fontCascade().letterSpacing())) {
```

Comment:
We usually prefer C++ style casts (static_cast<>) or conversion operators (!! for bool conversion). Here using a helper local (`auto shouldDisableLigatures == !!fontCascade().letterSpacing();`) would be more readable.